packages/core/src/ui/widgets/colorBar/ViewportColorBar.ts

```diff
@@ -0,0 +1,173 @@
+import {
+  eventTarget,
+  VolumeViewport,
+  StackViewport,
+  Types,
+  Enums,
+  utilities,
+  getEnabledElement,
+  cache,
+} from '@cornerstonejs/core';
+import { ColorBar } from './ColorBar';
+import { ColorBarProps, ColorBarVOIRange } from './types';
+
+const { Events } = Enums;
+const DEFAULT_MULTIPLIER = 4;
+
+export interface ViewportColorBarProps extends ColorBarProps {
+  element: HTMLDivElement;
+  volumeId?: string;
+}
+
+class ViewportColorBar extends ColorBar {
+  private _element: HTMLDivElement;
+  private _volumeId: string;
+
+  constructor(props: ViewportColorBarProps) {
+    super({
+      ...props,
+      range: ViewportColorBar._getRange(props.element, props.volumeId),
+      voiRange: ViewportColorBar._getVOIRange(props.element, props.volumeId),
+    });
+
+    this._element = props.element;
```

Comment:
also didn't see that element is part of the new ViewportColorBar props, I saw the viewportId, I actually like the viewportId approach more